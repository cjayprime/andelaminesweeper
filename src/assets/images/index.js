export {default as cancel} from './cancel.png';
export {default as explode} from './explode.png';
export {default as flag} from './flag.png';
export {default as mine} from './mine.png';
export {default as timer} from './timer.png';
export {default as reload} from './reload.png';
export {default as checked} from './checked.png';
export {default as shovel} from './shovel.png';
export {default as arrowDown} from './arrowDown.png';
