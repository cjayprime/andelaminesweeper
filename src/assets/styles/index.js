export { default as tiles } from './tiles';
export { default as board } from './board';
export { default as header } from './header';
export { default as button } from './button';
export { default as animation } from './animation';
export { default as modal } from './modal';
export { default as picker } from './picker';