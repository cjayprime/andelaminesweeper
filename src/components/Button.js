import React, {Component} from 'react';
import {TouchableOpacity, View, Text, Image} from 'react-native';

import * as Style from '../assets/styles';

import {Constants} from '../utils';

export default class Button extends Component {
  render() {
    const height = Constants.HEADER_HEIGHT;
    const {image, label, onPress} = this.props;
    return (
      <View style={Style.button.container}>
        <TouchableOpacity
          onPress={onPress}
          style={[Style.button.click, {height: 0.7 * height}]}>
          <Text style={Style.button.text}>{label}</Text>
          {image && <Image source={image} style={Style.button.image} />}
        </TouchableOpacity>
      </View>
    );
  }
}
