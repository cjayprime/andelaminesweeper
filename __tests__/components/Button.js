import React from 'react';
import Button from '../../src/components/Button';
import {arrowDown} from '../../src/assets/images';

import renderer from 'react-test-renderer';

jest.useFakeTimers();
describe('Button Test', () => {
    test('Stability', () => {
        const tree = renderer.create(
            <Button
                label={'Easy'}
                onPress={() => {}}
                image={arrowDown}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});