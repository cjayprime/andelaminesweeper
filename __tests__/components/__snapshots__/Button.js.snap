// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test Stability 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginTop": -10,
      "width": "30%",
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "borderRadius": 5,
        "elevation": 3,
        "flexDirection": "row",
        "height": 30.799999999999997,
        "justifyContent": "space-around",
        "marginTop": 10,
        "opacity": 1,
        "width": "85%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 15,
        }
      }
    >
      Easy
    </Text>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/arrowDown.png",
        }
      }
      style={
        Object {
          "height": 15,
          "width": 15,
        }
      }
    />
  </View>
</View>
`;
